### PR TITLE
Remove Winndos warnings for lib source files 

### PR DESCRIPTION
**Pull request by lucianomartin**
I will remove the warnings in the tests and demos in a separate PR.

Part of https://xmosjira.atlassian.net/browse/AP-179

### CODE REVIEW

File: lib_xcore_math/src/bfp/bfp_alloc.c [hunk 1]
```diff
@@ -11,7 +11,7 @@
 bfp_s32_t bfp_s32_alloc(
     const unsigned length)
 {
-#if (DEBUG)
+#if (defined DEBUG && DEBUG > 0)
```
Comment (uvvpavel) [inline lib_xcore_math/src/bfp/bfp_alloc.c:14]: Is this config even being defined?

**Review by uvvpavel — COMMENTED** [review]
As far as I saw we still have -Wextra and -Werror not applied to the windows build. Should we enable them back?

File: lib_xcore_math/src/bfp/bfp_alloc.c (hunk 1 above)
Comment (lucianomartin) [reply lib_xcore_math/src/bfp/bfp_alloc.c:14]: I thought that it can be given via CLI. It is not defined within the code.

**Comment by lucianomartin** [comment]
> As far as I saw we still have -Wextra and -Werror not applied to the windows build. Should we enable them back?

On windows -Wextra and -Werror  are not valid, at least on my CL compiler. Do you expect them to be used on Windows?

**Comment by uvvpavel** [comment]
> > As far as I saw we still have -Wextra and -Werror not applied to the windows build. Should we enable them back?
> 
> On windows -Wextra and -Werror are not valid, at least on my CL compiler. Do you expect them to be used on Windows?

I don't really see a point in cleaning up the warnings and not enabling the -Werror, they will come back and no one's gonna notice that

**Comment by lucianomartin** [comment]
> > > As far as I saw we still have -Wextra and -Werror not applied to the windows build. Should we enable them back?
> > 
> > 
> > On windows -Wextra and -Werror are not valid, at least on my CL compiler. Do you expect them to be used on Windows?
> 
> I don't really see a point in cleaning up the warnings and not enabling the -Werror, they will come back and no one's gonna notice that

good point. On Windows we can use `-WX` to treat all warnings as errors.
I will enable that flag in a future PR.

**Review by uvvpavel — APPROVED** [review]
All good!

**Pull request merged**